types/express-serve-static-core/index.d.ts

```diff
@@ -440,7 +440,10 @@ interface Request extends http.IncomingMessage, Express.Request {
 
     method: string;
 
-    params: any;
+    params: {
+        [namedParameter: string]: string;
+        [captureGroup: number]: string;
+    };
```

Comment:
@OliverJAsh Thanks, good to know. So `[captureGroup: number]` is only required for arrays? Or it's currently unused by the language?

It's weird but this example has no error too:
```ts
const a: number[] = [1, 2, 3]
a["0"] = 0
a["five"] = "five" // no error
```